src/System.IO.FileSystem/tests/FileStream/Dispose.cs

```diff
@@ -58,49 +58,67 @@ protected override void Dispose(bool disposing)
             }
         }
 
-        [ActiveIssue(6153, PlatformID.AnyUnix)]
         [Fact]
-        public void DisposeVirtualBehavior()
+        public void Dispose_CallsVirtualDispose_TrueArg()
         {
-            bool called = false;
+            bool disposeInvoked = false;
 
-            // Normal dispose should call Dispose(true)
-            using (MyFileStream fs = new MyFileStream(GetTestFilePath(), FileMode.Create))
+            Action act = () => // separate method to avoid JIT lifetime-extension issues
             {
-                fs.DisposeMethod = (disposing) =>
+                using (MyFileStream fs = new MyFileStream(GetTestFilePath(), FileMode.Create))
                 {
-                    called = true;
-                    Assert.True(disposing);
-                };
-
-                fs.Dispose();
-                Assert.True(called);
+                    fs.DisposeMethod = (disposing) =>
+                    {
+                        disposeInvoked = true;
+                        Assert.True(disposing, "Expected true arg to Dispose(bool)");
+                    };
+
+                    // Normal dispose should call Dispose(true)
+                    fs.Dispose();
+                    Assert.True(disposeInvoked, "Expected Dispose(true) to be called from Dispose()");
+
+                    disposeInvoked = false;
+                }
+
+                // Second dispose leaving the using should still call dispose
+                Assert.True(disposeInvoked, "Expected Dispose(true) to be called from Dispose() again");
+                disposeInvoked = false;
+            };
+            act();
 
-                called = false;
+            // Make sure we suppressed finalization
+            for (int i = 0; i < 2; i++)
+            {
+                GC.Collect();
+                GC.WaitForPendingFinalizers();
             }
+            Assert.False(disposeInvoked, "Expected finalizer to have been suppressed");
+        }
 
-            // Second dispose leaving the using should still call dispose
-            Assert.True(called);
-
-            called = false;
-            // make sure we suppress finalization
-            GC.Collect();
-            GC.WaitForPendingFinalizers();
-            Assert.False(called);
+        [Fact]
+        public void Finalizer_CallsVirtualDispose_FalseArg()
+        {
+            bool disposeInvoked = false;
 
-            // Dispose from finalizer should call Dispose(false)
-            called = false;
-            MyFileStream fs2 = new MyFileStream(GetTestFilePath(), FileMode.Create);
-            fs2.DisposeMethod = (disposing) =>
+            Action act = () => // separate method to avoid JIT lifetime-extension issues
             {
-                called = true;
-                Assert.False(disposing);
+                var fs2 = new MyFileStream(GetTestFilePath(), FileMode.Create)
+                {
+                    DisposeMethod = (disposing) =>
```

Comment:
Yeah, in the other test, it looked weird using a multi-line object initializer in the using, so I left that one as a normal property access.